### DISCUSSION TITLE
Deno runtime

### DISCUSSION THREAD

**Discussion by mfulton26** [community/community#160501] [Actions]
`actions/github-script` is helpful but I'd like to not hassle with NPM. Can Deno be offered as an alternative to Node for a JavaScript runtime for actions? Deno also supports TypeScript so this would enable that TypeScript support too.

**Comment by laughedelic** [comment]
This would simplify maintenance of the JS/TS-based actions a lot, because developers wouldn't need to build them on every release and check in the build artifacts in the git repo. Deno can run JS/TS on the fly and project setup/configuration is much (much!) simpler than for Node.js projects.

**Comment by flinn** [comment]
I would love to see this as well so that GitHub Actions/Workflows can benefit in terms of both TypeScript support and overall security 👍

**Comment by roryabraham** [comment]
We have a number of actions in a monorepo where checking in `node_modules` wouldn't be acceptable. So instead, we bundle our JS/TS actions with ncc. This works reasonably well, but:

- It's annoying having an extra build step
- Because we check in the bundled actions, they show up in code diffs and pull requests, which is just distracting for our team (and for AI review tools such as copilot)
